mathoxide-lib/src/shape_utils.rs

```diff
@@ -0,0 +1,58 @@
+pub enum ShapeDim {
+    Known(usize),
+    Dyn,
```

Comment:
I am to blame for this one but maybe `Inferred` would be a better name than `Dyn` since even `Known` values could be computed at runtime.